sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -144,10 +144,14 @@ object DataType {
 
     case JSortedObject(
     ("class", JString(udtClass)),
-    ("pyClass", _),
-    ("sqlType", _),
+    ("pyClass", pyClass: JValue),  // pyClass could be null, cannot match with JString()
+    ("sqlType", v: JValue),
     ("type", JString("udt"))) =>
-      Utils.classForName(udtClass).newInstance().asInstanceOf[UserDefinedType[_]]
+      if (udtClass.length > 0) {
+        Utils.classForName(udtClass).newInstance().asInstanceOf[UserDefinedType[_]]
+      } else {
+        new PythonUserDefinedType(parseDataType(v), pyClass.asInstanceOf[JString].values)
```

Comment:
I guess we only need UDT support in Python + JVM for now, but I wonder whether assuming that an empty string implies Python is the right long-term approach if we ever decide to support, say, R UDTs.